symmetric/src/hasher.rs

```diff
@@ -1,8 +1,26 @@
+/// A generic trait for cryptographic hashers that consume an arbitrary sequence of input items
+/// and produce a fixed-size output.
+///
+/// This trait abstracts over hash functions in a flexible way, supporting both field elements,
+/// scalars, or any other data type that implements `Clone`.
 pub trait CryptographicHasher<Item: Clone, Out>: Clone {
+    /// Hash an iterator of input items into a single digest output.    ///
+    /// # Arguments
+    /// - `input`: An iterator over items to be hashed.
+    ///
+    /// # Returns
+    /// A fixed-size digest of type `Out`.
     fn hash_iter<I>(&self, input: I) -> Out
     where
         I: IntoIterator<Item = Item>;
 
+    /// Hash an iterator of slices, automatically flattening them into a single stream of items.
```

Comment:
(Thought this rewording makes is clear that the flattening is a guarantee just in case someone comes up with a different way to hash a stream of slices) 
```suggestion
    /// Hash an iterator of slices, by flattening it into a single stream of items.
```